File: lib/metadata.ml
type metadata =
  { title : string
  ; album_artist : string
  ; album : string
  ; artists : string list
  }
[@@deriving show]

type t = metadata

let pp_metadata x = Printf.printf "%s\n" (show_metadata x)
let create () = { title = ""; album = ""; album_artist = ""; artists = [] }

File: src/metadata.ts
import { Schema } from "effect";

export const MetadataSchema = Schema.Struct({
	album: Schema.NonEmptyString,
	artist: Schema.NonEmptyString,
	title: Schema.NonEmptyString,
	trackNumber: Schema.optional(Schema.Number),
});

const mutableMetadataSchema = Schema.mutable(MetadataSchema);

export type Metadata = Schema.Schema.Type<typeof mutableMetadataSchema>;
